### ISSUE TITLE
Vim hangs with plugin enabled on Catalina

### ISSUE THREAD

**Issue by Gray-Wind** [codota/tabnine-vim#41]
# Issue Prelude

- [x] I have read and understood YCM's [CONTRIBUTING][cont] document.
- [x] I have read and understood YCM's [CODE_OF_CONDUCT][code] document.
- [x] I have read and understood YCM's [README][readme], **especially the
  [Frequently Asked Questions][faq] section.**
- [x] I have searched YCM's issue tracker to find issues similar to the one I'm
  about to report and couldn't find an answer to my problem. ([Example Google
  search.][search])
- [x] If filing a bug report, I have included the output of `vim --version`.
- [ ] If filing a bug report, I have included the output of `:YcmDebugInfo`.
- [ ] If filing a bug report, I have attached the contents of the logfiles using
  the `:YcmToggleLogs` command.
- [x] If filing a bug report, I have included which OS (including specific OS
  version) I am using.
- [x] If filing a bug report, I have included a minimal test case that reproduces
  my issue, including what I expected to happen and what actually happened.
- [x] **I understand this is an open-source project staffed by volunteers and
  that any help I receive is a selfless, heartfelt _gift_ of their free time. I
  know I am not entitled to anything and will be polite and courteous.**
- [x] **I understand my issue may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Issue Details

Vim hangs and do not react to any commands. CPU usage goes wild.

* What did you do?
Run vim with this minimal config:

```
filetype off

set rtp+=~/.vim/bundle/Vundle.vim/
call vundle#begin()

Plugin 'zxqfl/tabnine-vim'

call vundle#end()
filetype plugin indent on
```

# Diagnostic data

## Output of `vim --version`

```
vim --version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Sep  5 2019 21:03:34)
Included patches: 1-503, 505-680, 682-1312
Compiled by root@apple.com
Normal version without GUI.  Features included (+) or not (-):
+acl               +extra_search      -mouse_netterm     -tag_old_static
-arabic            -farsi             +mouse_sgr         -tag_any_white
+autocmd           +file_in_path      -mouse_sysmouse    -tcl
+autochdir         +find_in_path      -mouse_urxvt       -termguicolors
-autoservername    +float             +mouse_xterm       +terminal
-balloon_eval      +folding           +multi_byte        +terminfo
-balloon_eval_term -footer            +multi_lang        +termresponse
-browse            +fork()            -mzscheme          +textobjects
+builtin_terms     -gettext           +netbeans_intg     +textprop
+byte_offset       -hangul_input      +num64             +timers
+channel           +iconv             +packages          +title
+cindent           +insert_expand     +path_extra        -toolbar
-clientserver      +job               -perl              +user_commands
-clipboard         +jumplist          +persistent_undo   -vartabs
+cmdline_compl     -keymap            +postscript        +vertsplit
+cmdline_hist      +lambda            +printer           +virtualedit
+cmdline_info      -langmap           -profile           +visual
+comments          +libcall           +python/dyn        +visualextra
-conceal           +linebreak         -python3           +viminfo
+cryptv            +lispindent        +quickfix          +vreplace
+cscope            +listcmds          +reltime           +wildignore
+cursorbind        +localmap          -rightleft         +wildmenu
+cursorshape       -lua               +ruby/dyn          +windows
+dialog_con        +menu              +scrollbind        +writebackup
+diff              +mksession         +signs             -X11
+digraphs          +modify_fname      +smartindent       -xfontset
-dnd               +mouse             +startuptime       -xim
-ebcdic            -mouseshape        +statusline        -xpm
-emacs_tags        -mouse_dec         -sun_workshop      -xsmp
+eval              -mouse_gpm         +syntax            -xterm_clipboard
+ex_extra          -mouse_jsbterm     +tag_binary        -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L/usr/local/lib -o vim        -lm -lncurses  -liconv -framework Cocoa
```

## Output of `YcmDebugInfo`

> Not able to gather

## Contents of YCM, ycmd and completion engine logfiles

> Not able to gather

## OS version, distribution, etc.

> Mac OS Catalina 10.15

[cont]: https://github.com/Valloric/YouCompleteMe/blob/master/CONTRIBUTING.md
[code]: https://github.com/Valloric/YouCompleteMe/blob/master/CODE_OF_CONDUCT.md
[readme]: https://github.com/Valloric/YouCompleteMe/blob/master/README.md
[faq]: https://github.com/Valloric/YouCompleteMe/blob/master/README.md#faq
[search]: https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fgithub.com%2FValloric%2FYouCompleteMe%2Fissues%20python%20mac
[gist]: https://gist.github.com/

**Comment by oleg578** [comment]
I confirm also

**Comment by aloyr** [comment]
same here. got vim to work by temporarily removing the YCM plugin.

adding vim and python to accessibility, input monitoring, and full disk access under _preferences => security & privacy => privacy_ did nothing.

**Comment by halilim** [comment]
https://github.com/ycm-core/YouCompleteMe/issues/3520 is apparently the same issue on YCM side (which I can confirm).

**Comment by MrMonk3y** [comment]
> [ycm-core#3520](https://github.com/ycm-core/YouCompleteMe/issues/3520) is apparently the same issue on YCM side (which I can confirm).

It's exactly the same. Sorry haven't seen this one.

**Comment by YigitDemirag** [comment]
This issue still persists on vim and macvim..

**Comment by jpp46** [comment]
This is exactly what happens to me as well on macOS Catalina

**Comment by thanhhao98** [comment]
Is there solution for this issue?

**Comment by MrMonk3y** [comment]
updating macvim worked for me and most others [https://github.com/ycm-core/YouCompleteMe/issues/3520#issuecomment-547920930](https://github.com/ycm-core/YouCompleteMe/issues/3520#issuecomment-547920930)

**Comment by trevvvy** [comment]
I finally was able to solve this issue. In my case, if I included the tabnine-vim plugin in my vimrc, both vim and Macvim would freeze immediately upon opening with sustained 100% CPU usage.

Here's what did the trick:

1. uninstalled all versions of python i had accumulated in brew
2. trashed my downloaded binary of MacVim
3. did a fresh install of vim (or macvim if that's your pref) with brew

My theory on why this fixed things is that the version of vim that ships with MacOS is not compatible with the you-complete-me code, which this project is based on.  Installing the latest brew version of vim, which installed python 3.9, seemed to make everything happy.

Hope this might help.
